### ISSUE TITLE
Docs: ddd for this project

### ISSUE THREAD

**Issue by toorusr** [CodeDesignInitiative/campy#8]
# Idea
I think in such complex projects it is really handy to use documentation driven development.

# What is ddd exactly?
Documentation driven development is the process of writing the documentation before writing the code.
It would be a good solution for this project because:
* everyone should understand what is going on
* code is better understandable
* the whole project would be very good documented
* ddd keeps up the motivation of developing

Some downsides to this are:
* time consuming
* confusing if not done correctly

# What do you think?
We can discuss about this together under this issue.

# Reference
[ddd explanation](http://www.thinkingphp.org/spliceit/docs/0.1_alpha/pages/ddd_info.html)
###### optional:
[rdd - readme driven development](https://github.com/mojombo/mojombo.github.io/blob/master/_posts/2010-08-23-readme-driven-development.md)

**Comment by minthemiddle** [comment]
Good idea. Actually, I think a good mix of `documentation driven development (ddd)` and `test driven development (tdd)` would be the best for this project. The code needs to be stable yet fast moving and we would love to have as many contributors as possible. Documenting is a lot easier than writing tests (also for me, especially with lots of non-programming tasks on the desk…).

**Comment by toorusr** [comment]
It totally agree with you there.
